src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,142 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private int size;
+    private int capacity;
+    private Object[] elements;
+
+    public ArrayList() {
+        capacity = DEFAULT_CAPACITY;
+        elements = new Object[DEFAULT_CAPACITY];
```

Comment:
```suggestion
        elements = (T[]) new Object[DEFAULT_CAPACITY];
```
just a suggestion, it lets you avoid unnecessary casting